### PR TITLE
Added coinChange algorithm in javascript by I3han

### PR DESCRIPTION
**Pull request by I3han**
**Hi!**  
Thanks for considering for contributing to this ever-growing list of algorithm and data structure implementations.
Your contribution is valuable. Use the follow template for creating PRs. (Delete this part)
---

**Coin Change Problem in Javascript #571**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

**Comment by RavinduSachintha** [comment]
@I3han the file name is not in the expected convention. first, you need to take a look at the contributing doc. then you will able to find out the mistakes.

File: algorithms/ar-cochprb/javascript/coin_change_problem.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    
+    if(numOfCoins < 0 && n > 0) return 0; 
+
+    return waysofChange(s, numOfCoins, n - s[numOfCoins]) + 
+    waysofChange(s, numOfCoins - 1, n);
+}
+
+var s = [1, 2, 3];
+var n = 4;
+console.log(waysofChange(s, s.length - 1, n));
```
Comment (RavinduSachintha) [inline algorithms/ar-cochprb/javascript/coin_change_problem.js:16]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6356108)

File: algorithms/ar-cochprb/javascript/coin_change_problem.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+function waysofChange(s, numOfCoins, n) {
+    
+    if(n === 0) return 1; 
+    
+    if(n < 0) return 0; 
+    
+    if(numOfCoins < 0 && n > 0) return 0;
```
Comment (RavinduSachintha) [inline algorithms/ar-cochprb/javascript/coin_change_problem.js:8]: Codacy found an issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6356108)

File: algorithms/ar-cochprb/javascript/coin_change_problem.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+function waysofChange(s, numOfCoins, n) {
+    
+    if(n === 0) return 1;
```
Comment (RavinduSachintha) [inline algorithms/ar-cochprb/javascript/coin_change_problem.js:4]: Codacy found an issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6356108)

File: algorithms/ar-cochprb/javascript/coin_change_problem.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+
+function waysofChange(s, numOfCoins, n) {
+    
+    if(n === 0) return 1; 
+    
+    if(n < 0) return 0;
```
Comment (RavinduSachintha) [inline algorithms/ar-cochprb/javascript/coin_change_problem.js:6]: Codacy found an issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6356108)

File: algorithms/ar-cochprb/javascript/coin_change_problem.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    if(numOfCoins < 0 && n > 0){
+        return 0; 
+    }
+
+    return waysofChange(s, numOfCoins, n - s[numOfCoins]) + waysofChange(s, numOfCoins - 1, n);
+}
+
+var s = [1, 2, 3];
+var n = 4;
+console.log(waysofChange(s, s.length - 1, n));
```
Comment (RavinduSachintha) [inline algorithms/ar-cochprb/javascript/coin_change_problem.js:21]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6356108)

Comment (I3han) [reply algorithms/ar-cochprb/javascript/coin_change_problem.js:21]: any idea how to fix this?

**Comment by I3han** [comment]
@RavinduSachintha can u please check about issues

File: algorithms/ar-cochprb/javascript/coin_change_problem.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        return 0;
+    }
+
+    return waysofChange(s, numOfCoins, n - s[numOfCoins]) + waysofChange(s, numOfCoins - 1, n);
+}
+
+function main() {
+    var s = [1, 2, 3];
+    var n = 4;
+    console.error(waysofChange(s, s.length - 1, n));
```
Comment (RavinduSachintha) [inline algorithms/ar-cochprb/javascript/coin_change_problem.js:22]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6356108)

**Review by RavinduSachintha — CHANGES_REQUESTED** [review]
@I3han you need to remove the `.inode/symbolCache.db` file from the PR. After that, please add this contribution to the algorithm chart on the readme file. don't change anything else. :+1:

**Comment by I3han** [comment]
@RavinduSachintha contribution is already added to the algorithm chart

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@all-contributors please add @I3han for code

**Comment by RavinduSachintha** [comment]
@I3han thank you for your contribution. keep it up :+1: 
make sure to add a star to this repo :wink:

**Pull request merged**